core/models.py

```diff
@@ -28,7 +30,21 @@ def create_for_url(self, url, referer=None):
         if referer is not None:
             headers["Referer"] = referer
         response = requests.get(url, headers=headers)
-        buf.write(response.content)
+        if response.content[:4] == b'<svg':
+            imageType = 'svg'
+            svg2png(bytestring=response.content,write_to='output')
+
+            buf.write(open('output','rb').read())
+        else:
+            buf.write(response.content)
+            tempImage = open('output','wb')
+            tempImage.write(response.content)
+            tempImage.close()
+
+        dims = PILImage.open('output').size
+        if dims[0] < 45 or dims[1] < 45:
+            ##Write here logic for clean exit if image is small like icons.
```

Comment:
How did you choose 45x45 to be size of the icons?

Reply:
I saw many images and came to conclusion, but the "many" may not be sufficient will need to build a better metric for this classification.